File: 205/solution.c
#include "../tools/pe.h"
#include <time.h>


gint
roll(num_dice, num_sides)
{
    gint sum, i;

    sum = 0;
    for(i = 0; i < num_dice; i++)
    {
        sum += (rand() % num_sides) + 1;
    }
    return sum;
}
    
    
gint
main()
{
    gint i, j, peter, colin;
    gfloat win, total;

    srand(time(NULL));
    peter, colin, total = 0;

    win = 0;
    total = 0;

    // 0.57315066
    // 0.5732091
    for(i = 0; i <= 10000000; i++)
    {
        for(j = 0; j <= 1000; j++)
        {
            peter = roll(9, 4);
            colin = roll(6, 6);
            if(peter > colin)
                win++;
            total++;
        }
    }
    printf("Wins: %.0f\tTotal: %.0f\n", win, total);
    printf("Win percentage = %0.7f\n", win / total);

    return 0;
}

File: 205/solution.rb
#!/usr/bin/env ruby
#
# projecteuler.net
#
# problem 205
#
# Peter has nine four-sided (pyramidal) dice, each with faces numbered
# 1, 2, 3, 4.
# Colin has six six-sided (cubic) dice, each with faces numbered 1, 2,
# 3, 4, 5, 6.
#
# Peter and Colin roll their dice and compare totals: the highest total
# wins. The result is a draw if the totals are equal.
#
# What is the probability that Pyramidal Pete beats Cubic Colin? Give
# your answer rounded to seven decimal places in the form 0.abcdefg
#


def roll(num_dice, num_sides)
    sum = 0
    num_dice.times do |die|
        sum += rand(num_sides) + 1
    end
    sum
end

total = 0
win = 0
5000000.times do
    peter = roll(9,4)
    colin = roll(6,6)
    win += 1 if peter > colin
    total += 1
end

printf("Win percentage = %0.7f\n", win / total.to_f)


#  printf("%0.7f\n", 10/3.0)


